core/src/main/java/org/jdbi/v3/core/mapper/EnumByNameMapperFactory2.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.mapper;
+
+import java.lang.reflect.Type;
+import java.util.Optional;
+import org.jdbi.v3.core.EnumByName;
+import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.generic.GenericTypes;
+
+/**
+ * Produces enum column mappers, which case-insensitively map enums from textual columns according to their {@link Enum#name()}.
+ */
+@EnumByName
+class EnumByNameMapperFactory2 implements ColumnMapperFactory {
```

Comment:
I think we'd be better served by refactoring `EnumMapperFactory` (which is package private) to implement `QualifiedColumnMapperFactory`, and follow the same logic I suggested for `EnumArgumentFactory`. Then these new classes are redundant.